yolopass-backend/app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Admin::UsersController < ApplicationController
+  before_action :authorize_admin
+
+  def index
+    users = User.all
+    render json: users
+  end
+
+  #cannot delete admin
+  def destroy
+    user = User.find(params[:id])
+  
+    if user.admin?
+      render json: { error: 'Cannot delete an admin' }, status: :unprocessable_entity
```

Comment:
@PranaliM0 Messages should be mapped using [Internationalization](https://guides.rubyonrails.org/i18n.html)